### PR TITLE
[FEATURE] Make records timeable and hidable

### PR DESCRIPTION
**Pull request by lukaszuznanski**
Resolves: #553 

Tested on v12 and v11

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 - Add an FE editor (#864, #872, #874, #876)
 - Add automerging of green Dependabot PRs (#756)
 - Add type coverage calculation (#830)
+- Make records timeable and hideable (#553)
```
Comment (oliverklee) [inline CHANGELOG.md:13]: This line should go on top so the changelog is in reverse chronological order.

File: Configuration/TCA/tx_tea_domain_model_product_tea.php [hunk 2]
```diff
@@ -24,16 +27,64 @@
...
+                endtime;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:endtime_formlabel,
+                --linebreak--,
+                fe_group;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:fe_group_formlabel,
+            ',
         ],
     ],
     'columns' => [
+        'hidden' => [
+            'exclude' => true,
+            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.enabled',
```
Comment (oliverklee) [inline Configuration/TCA/tx_tea_domain_model_product_tea.php:53]: This label doesn't quit match with teas: In German, the labels says "Sichtbarkeit des Inhaltelemens" (translated: "visibility of the content element"). As tea records are no content elements, we probably will need to have our own label for this.

**Review by oliverklee — CHANGES_REQUESTED** [review]
I've manually tested this in a 11LTS and 12LTS backend: It works fine for me in both version. Thanks!

We should change a label, though.

File: CHANGELOG.md (hunk 1 above)
Comment (oliverklee) [reply CHANGELOG.md:13]: Oh, and also we should use the new PR number #989.

File: Configuration/TCA/tx_tea_domain_model_product_tea.php (hunk 2 above)
Comment (oliverklee) [reply Configuration/TCA/tx_tea_domain_model_product_tea.php:53]: My proposal for the label:
- "Visible" (EN)
- "Sichtbar" (DE)

File: Resources/Private/Language/de.locallang_db.xlf [hunk 3]
```diff
@@ -31,6 +31,9 @@
 				<source>Usergroup Access Rights</source>
 				<target>Zugriffsrechte für Benutzergruppen</target>
 			</trans-unit>
+			<trans-unit id="tx_tea_domain_model_product_tea.hidden">
+				<source>Sichtbar</source>
```
Comment (oliverklee) [inline Resources/Private/Language/de.locallang_db.xlf:35]: This needs to be the `target` (with the `source` copied from the `locallang_db.xml`).

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: Resources/Private/Language/de.locallang_db.xlf (hunk 3 above)
Comment (lukaszuznanski) [reply Resources/Private/Language/de.locallang_db.xlf:35]: Yea, my fault, fixing right now.

**Comment by oliverklee** [comment]
Hm, this is strange. With the latest version (on another machine, though), I still see the "visibility of content element" label in V11, and don't see the new fields in the BE in V12 at all.

File: Configuration/TCA/tx_tea_domain_model_product_tea.php [hunk 4]
```diff
@@ -24,16 +27,64 @@
...
+            'l10n_mode' => 'exclude',
+            'l10n_display' => 'defaultAsReadonly',
+        ],
+        'endtime' => [
+            'exclude' => true,
+            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.endtime',
+            'config' => [
+                'type' => 'datetime',
+                'default' => 0,
+                'range' => [
```
Comment (oliverklee) [inline Configuration/TCA/tx_tea_domain_model_product_tea.php:81]: Why is there an upper limit for the endtime, but not for the starttime?

Comment (lukaszuznanski) [reply Configuration/TCA/tx_tea_domain_model_product_tea.php:81]: I've taken it from tt_content from core, should we adjust it and also fix it for core?

**Review by oliverklee — APPROVED** [review]
Works fine on my V11 and V12 installations. Thanks!

**Pull request merged**